gh-badges/lib/index.js

```diff
@@ -1,6 +1,29 @@
 'use strict'
 
+const c = require('rho-contracts-fork')
 const makeBadge = require('./make-badge')
+const schema = c.object({
+  text: c.tuple(c.string, c.string).strict(),
+  labelColor: c.optional(c.string),
+  color: c.optional(c.string),
+  colorA: c.optional(c.string),
+  colorscheme: c.optional(c.string),
+  colorB: c.optional(c.string),
+  format: c.optional(c.oneOf('svg', 'json')),
```

Comment:
I thought this accepted raster formats too. Am I wrong about that? Do we have an API for making raster formats or are they only available in the CLI?

Reply:
It doesn't. At the moment, if you pass `format: 'gif'` (or `'png'` or `'jpg'`), you just get a SVG badge back:

```js
> const { BadgeFactory } = require('./gh-badges')
undefined

> const bf = new BadgeFactory()
undefined

> bf.create({text: ['foo', 'bar']})
'<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="56" height="20"><linearGradient id="b" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="a"><rect width="56" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#a)"><path fill="#555" d="M0 0h27v20H0z"/><path fill="#4c1" d="M27 0h29v20H27z"/><path fill="url(#b)" d="M0 0h56v20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="110"> <text x="145" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="170">foo</text><text x="145" y="140" transform="scale(.1)" textLength="170">foo</text><text x="405" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="190">bar</text><text x="405" y="140" transform="scale(.1)" textLength="190">bar</text></g> </svg>'

> bf.create({text: ['foo', 'bar'], format: 'json'})
'{"label":"foo","message":"bar","link":["",""],"name":"foo","value":"bar"}'

> bf.create({text: ['foo', 'bar'], format: 'png'})
'<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="56" height="20"><linearGradient id="b" x2="0" y2="100%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><clipPath id="a"><rect width="56" height="20" rx="3" fill="#fff"/></clipPath><g clip-path="url(#a)"><path fill="#555" d="M0 0h27v20H0z"/><path fill="#4c1" d="M27 0h29v20H27z"/><path fill="url(#b)" d="M0 0h56v20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="110"> <text x="145" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="170">foo</text><text x="145" y="140" transform="scale(.1)" textLength="170">foo</text><text x="405" y="150" fill="#010101" fill-opacity=".3" transform="scale(.1)" textLength="190">bar</text><text x="405" y="140" transform="scale(.1)" textLength="190">bar</text></g> </svg>'
```

The code to do raster formats is only in the CLI wrapper:

https://github.com/badges/shields/blob/c312eae3e3cfa28d8442b9e691ff063a85b3e3e5/gh-badges/lib/badge-cli.js#L69-L78

We could easily add that feature to the library too, which would be more consistent with the CLI. Simultaneously, we're generally trying to move away from raster format support so I'm not sure we want to introduce that here given we've either never supported it (or possibly we removed it and nobody cared).

Reply:
Have you had any further thoughts on this in the context of the work you've been doing on moving raster support out to another service? I'd quite like to get this settled one way or the other so we can finish this PR off and have a look at the other v3 roadmap tasks..